src/js/stores/MesosStateStore.js

```diff
@@ -44,15 +44,15 @@ function stopPolling() {
 /**
  * Assigns a property to task if it is a scheduler task.
  * @param  {Object} task
- * @param  {Array} schedulerTasks Array of scheduler task
+ * @param  {Object} schedulerTasksMap Map of scheduler task
  * @return {Object} task
  */
-function assignSchedulerTaskField(task, schedulerTasks) {
-  if (schedulerTasks.some(({ id }) => task.id === id)) {
-    return Object.assign({}, task, { schedulerTask: true });
+function assignSchedulerTaskField(task, schedulerTasksMap) {
+  if (schedulerTasksMap[task.id] == null) {
```

Comment:
Double equals always make me a bit nervous, I think I personally would prefer to do `!schedulerTasksMap[task.id]`, but it's okay for me like this 🤷‍♂️

Reply:
It doesn't really matter in this case as we only need to check for `undefined`. I went with `value == null` to relax the condition also catch cases in which some one `null`fied the value. Please note that `value == null` checks for `null` and `undefined` and `!value` for all falsey values.